### PR TITLE
documentation updates

### PR DESCRIPTION
**Pull request by dae**
The changes in DEVELOPING.md address some of the confusion I had when I first started using these rules

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -2,15 +2,62 @@
 
 We strongly encourage you to review the project's scope described in the `README.md` file before working on new features. For large changes, consider writing a design document using [this template](https://goo.gl/YCQttR).
 
-## Testing locally
+## Architecture
 
-This repository contains nested workspaces which are tested with the bazel-in-bazel bazel_integration_test rule. The integration tests must be run in series as they use up too many resources when run in parallel.
+Unlike some other Bazel rules, these rules require a separate build step, and
```
Comment (mattem) [inline DEVELOPING.md:7]: This is possible, see the [from_source](https://github.com/bazelbuild/rules_nodejs/tree/stable/examples/from_source) example, I'll agree though it's not intuitive of how to set it up.

File: docs/install.md [hunk 2]
```diff
@@ -157,3 +157,22 @@ Toolchains allow us to support cross-compilation, e.g. building a linux binary f
 So if for example you want to build a docker image from a non-linux platform you would run `bazel build --platforms=@build_bazel_rules_nodejs//toolchains/node:linux_amd64 //app`, which will ensure that the linux nodejs binary is downloaded and provided to the nodejs_binary target.
 
 Note: The toolchain currently only provides a platform-specific nodejs binary. Any native modules will still be fetched/built, by npm/yarn, for your host platform, so they will not work on the target platform. Support for cross-compilation with native dependencies will follow.
+
+Because these rules use the target platform to decide which node binary to use, you will run into trouble if you are trying to invoke these rules as part of a cross-compilation
```
Comment (mattem) [inline docs/install.md:161]: Perhaps we should add something to /examples for this?

**Review by alexeagle — APPROVED** [review]
Thanks for documentation contribution, that's super helpful

File: docs/install.md (hunk 2 above)
Comment (alexeagle) [reply docs/install.md:161]: filed https://github.com/bazelbuild/rules_nodejs/issues/2606 as a reminder so I can merge this

**Pull request merged**